modules/openapi-generator/src/test/resources/3_0/enum-implements.yaml

```diff
@@ -0,0 +1,33 @@
+openapi: 3.0.0
+info:
+  title: Sample API
+  description: API description in Markdown.
+  version: 1.0.0
+paths:
+  /ponies:
+    get:
+      summary: Returns all animals.
+      description: Optional extended description in Markdown.
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Pony'
+components:
+  schemas:
+    Pony:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/Type'
+    Type:
+      type: string
+      x-implements: org.openapitools.java.EnumInterface
```

Comment:
tested with this spec but it won't compile 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project openapi-java-client: Compilation failure
[ERROR] C:\Users\wing3\AppData\Local\Temp\javax\src\main\java\org\openapitools\client\model\Type.java:[30,49] error: cannot find symbol
[ERROR]   symbol:   class EnumInterface
[ERROR]   location: package org.openapitools.java

```

regardless it shows the PR works as expected.

ideally the test spec should generate code that compiles.

i'll get this PR merged first. if later you or someone can come up with a test spec generating code that compiles, that would be great.